toolchain/check/handle_impl.cpp

```diff
@@ -217,7 +217,9 @@ static auto PopImplIntroducerAndParamsAsNameComponent(
 
   Parse::NodeId first_param_node_id =
       context.node_stack().PopForSoloNodeId<Parse::NodeKind::ImplIntroducer>();
-  Parse::NodeId last_param_node_id = end_of_decl_node_id;
+  // Subtracting 1 since we don't want to include the final `{` or `;` of the
+  // declaration when performing syntactic match.
+  Parse::NodeId last_param_node_id(end_of_decl_node_id.index - 1);
```

Comment:
Optional: what do you think about a `CHECK` that the final token we're excluding is in fact `{` or `;`?